src/java/org/apache/cassandra/index/sai/utils/RowAwarePrimaryKeyFactory.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.utils;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.db.Clustering;
+import org.apache.cassandra.db.ClusteringComparator;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.dht.Token;
+import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+
+/**
+ * A row-aware {@link PrimaryKey.Factory}. This creates {@link PrimaryKey} instances that are
+ * sortable by {@link DecoratedKey} and {@link Clustering}.
+ */
+public class RowAwarePrimaryKeyFactory implements PrimaryKey.Factory
```

Comment:
I understand this comes from a previous version where we didn't have row-aware primary keys. Now this is a case of an interface with a single known implementation. Unless we have plans to add a partition-aware implementation or something else we should probably replace the interface by this class. wdyt?

The same would apply for `PrimaryKey`/`RowAwarePrimaryKey`.

Reply:
This is a good point. We are unlikely to ever introduce a `PartitionAwarePrimaryKey` or factory so both of these interfaces can go and be replaced with `PrimaryKey` and `PrimaryKeyFactory`.